### PR TITLE
improve readyness checks

### LINKED ISSUE #338: Improve readyness checks
Currently the app starts and listen for requests before having fully completed the initial catalog loading.  
This leads to `HTTP 500` errors when (re)deploying the app as Kubernetes routes requests to a not fully functionnal Onyxia-API.

### PR DESCRIPTION
**Pull request by ihiverlet**
close #338

### CODE REVIEW

**Review by johnksv — COMMENTED** [review]
Good initiative! I have been thinking about this my self but not had the time to do something about it.

My suggestion is to implement the check in `HealthcheckController`, as this is the endpoint that is used by kubernetes (both for the liveness and readiness probes, ref the [helm-chart](https://github.com/InseeFrLab/onyxia/blob/main/helm-chart/templates/deployment-api.yaml#L46)) to evaluate when the application is ready to receive traffic.

**Review by olevitt — APPROVED** [review]

**Comment by olevitt** [comment]
LGTM :+1: 
I suggest we wait for #362 and https://github.com/InseeFrLab/onyxia/issues/737 to be merged before merging this one as enabling readyness check with default 30sec deadline can lead to crashloopbackoffs

**Review by odysseu — APPROVED** [review]

File: onyxia-api/src/main/java/fr/insee/onyxia/api/controller/pub/HealthcheckController.java
Comment (johnksv) [inline onyxia-api/src/main/java/fr/insee/onyxia/api/controller/pub/HealthcheckController.java:19]: Side topic: I guess with this Onyxia also takes an position to move away from Kotlin (and the very limited part of kotlin code that exists) towards Java? If yes kotlin can probably be cleaned up from the build config, just so collaborators don't have to be unsure which language to use.

File: onyxia-api/src/main/java/fr/insee/onyxia/api/controller/pub/HealthcheckController.java
Comment (johnksv) [inline onyxia-api/src/main/java/fr/insee/onyxia/api/controller/pub/HealthcheckController.java:26]: Great seeing this added in the health check. If it is not to much to ask for I think a unit test covering some common scenarios would be very useful to verify that this code work as intended. Both now and in the future if/when changes will occur.

File: onyxia-api/src/main/java/fr/insee/onyxia/api/dao/universe/CatalogRefresher.java
Comment (olevitt) [inline onyxia-api/src/main/java/fr/insee/onyxia/api/dao/universe/CatalogRefresher.java:56]: Now that you chose the `healthcheck` solution, this `@PostConstruct` method is not suitable anymore as it would do double refresh and refresh before accepting network connections

**Review by olevitt — CHANGES_REQUESTED** [review]

**Comment by olevitt** [comment]
> LGTM 👍 I suggest we wait for #362 and [InseeFrLab/onyxia#737](https://github.com/InseeFrLab/onyxia/issues/737) to be merged before merging this one as enabling readyness check with default 30sec deadline can lead to crashloopbackoffs

PR mentionned above have been merged so green light for this one.
@ihiverlet Just need to take care of Sonarcloud remarks and cleanup the `@postconstruct` and we will be good to merge :+1:

File: onyxia-api/src/main/java/fr/insee/onyxia/api/controller/pub/HealthcheckController.java
Comment (johnksv) [inline onyxia-api/src/main/java/fr/insee/onyxia/api/controller/pub/HealthcheckController.java:23]: Use constructor based injection, instead of field based. This is described on sonarcloud (so I will not repeate it): https://sonarcloud.io/project/issues?resolved=false&pullRequest=353&id=InseeFrLab_onyxia-api&open=AY1ayCTVgs7rU-KXZaPs&tab=why

File: onyxia-api/src/main/java/fr/insee/onyxia/api/controller/pub/HealthcheckController.java
Comment (johnksv) [inline onyxia-api/src/main/java/fr/insee/onyxia/api/controller/pub/HealthcheckController.java:32]:
```suggestion
        if (uninitializedCatalogs.isEmpty()) {
```
I think isEmpty communicates better than `!= 0`.

**Review by olevitt — APPROVED** [review]
:+1:

**Review by johnksv — APPROVED** [review]

**Pull request merged**
